msal_extensions/persistence.py

```diff
@@ -149,9 +151,16 @@ def save(self, content):
         self._file_persistence.touch()  # For time_last_modified()
 
     def load(self):
+        """Return content from OSX Keychain, or None if the content does not exist."""
+        from .osx import KeychainError
         with self._Keychain() as locker:
-            return locker.get_generic_password(
-                self._service_name, self._account_name)
+            try:
+                return locker.get_generic_password(
+                    self._service_name, self._account_name)
+            except KeychainError as kce:
+                if kce.exit_status == KeychainError.ITEM_NOT_FOUND:
+                    return None  # Map ITEM_NOT_FOUND exception to None return value
```

Comment:
Yes, this is part of the happy path scenario. If you install AzCLI on Mac, you have no cache. MSAL will try to read it, but there is nothing there.